core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -190,8 +189,48 @@ private CodecFactory codec() {
       // add the Iceberg schema to keyValueMetadata
       meta("iceberg.schema", SchemaParser.toJson(schema));
 
+      Context context = createContextFunc.apply(config);
+      CodecFactory codec = context.codec();
+
       return new AvroFileAppender<>(
-          schema, AvroSchemaUtil.convert(schema, name), file, writerFunc, codec(), metadata, metricsConfig, overwrite);
+          schema, AvroSchemaUtil.convert(schema, name), file, writerFunc, codec, metadata, metricsConfig, overwrite);
+    }
+
+    private static class Context {
```

Comment:
Nit:  This class `Context` is private static class right, so I think all the following public methods can be marked as package-private, right ?

Reply:
Sure, let me update that.

Reply:
Fixed.